rir/src/runtime/DispatchTable.h

```diff
@@ -67,18 +67,31 @@ struct DispatchTable
         assert(fun->signature().optimization !=
                FunctionSignature::OptimizationLevel::Baseline);
         auto assumptions = fun->signature().assumptions;
-        assert(size() < capacity());
         size_t i = 1;
         for (; i < size(); ++i) {
             if (get(i)->signature().assumptions == assumptions) {
                 setEntry(i, fun->container());
                 return;
             }
-            if (!(assumptions < get(i)->signature().assumptions)) {
+            if (!(get(i)->signature().assumptions < assumptions)) {
                 break;
             }
         }
-        SLOWASSERT(!contains(fun->signature().assumptions));
+        assert(!contains(fun->signature().assumptions));
+        if (size() == capacity()) {
+            std::cout << "Tried to insert into a full Dispatch table. Have: \n";
+            for (size_t i = 0; i < size(); ++i) {
+                auto e = getEntry(i);
+                std::cout << "* "
+                          << Function::unpack(e)->signature().assumptions
+                          << "\n";
+            }
+            std::cout << "\n";
+            std::cout << "Tried to insert: " << assumptions << "\n";
+            Rf_error("failed");
```

Comment:
When should we start thinking about growing the dtable?

Reply:
for now I kind of like it when we get notified that we hit the current limit in the tests, because usually it points to some sort of error, rather than real version explosion...